UI refinements

nckrtl opened the pull request:
This PR contains a lot of little refinements for the ui as discussed in https://github.com/barryvdh/laravel-debugbar/issues/1593. As I was going through the queries tab and applying changes for the spacings I noticed I couldn't let the other tabs untouched as the changes also affected those.

So I went through all the tabs and made sure everything is well aligned there as well. This results in the PR being a bit hard to review by looking at the CSS changes. Probably better to checkout the branch and compare side by side. I know you mentioned to also apply changes in the base package if applicable but I haven't checked that yet. If you like the changes in this PR I could take another look to see if anything from this PR could be extracted to the base package.

Below are screenshots of dark mode but I made sure light mode also renders correctly.

### 

<img width="1488" alt="Screenshot 2024-04-01 at 23 07 49" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/4ba8bdc1-a9bd-4b91-92ea-bdc1c9aed5a6">

### 
The percentages bars were being displayed with table borders but I thought it would make sense to diplay them in light/dark background as the lines in Requests tabs are displayed as well. Making it a bit more consistent.

<img width="1487" alt="Screenshot 2024-04-01 at 23 08 08" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/60f5c6bb-3d0e-4c21-b286-9b30cb1e1d80">

### 
Updated spacing and line height of the lines in the error for bit better readability

<img width="1486" alt="Screenshot 2024-04-01 at 23 08 56" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/fbb56e0b-81d0-4035-8ec9-13fe459c10be">

### 

<img width="1484" alt="Screenshot 2024-04-01 at 23 09 33" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/910bc121-9c11-4fac-aff7-a1ec13338048">

### 

<img width="1488" alt="Screenshot 2024-04-01 at 23 09 47" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/35a5243b-301d-4606-8bcc-907e8910d370">

### 
I changed the color of the 1px line of the timeline measurement to blue to match the color that is being used on the timeline tab, felt more consistent like this. Also line height has been increased slightly for the table that contains the backtrace. Probably best to compare side by side

<img width="1487" alt="Screenshot 2024-04-01 at 23 10 12" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/a2dc280f-cabb-4718-a084-c7be1f279798">

### 
On the requests tab the background of the key items was not extending to the bottom of the of value items. This is now fixed and look a bit cleaner I think.

<img width="1488" alt="Screenshot 2024-04-01 at 23 10 47" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/4c7dfb98-9a62-4da1-affd-7f63aa5dd206">

### 
Adjust colours slightly and the table is moved up slightly to reduce the gap between the filters and the table

<img width="1486" alt="Screenshot 2024-04-01 at 23 11 03" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/be323942-12af-426d-abbe-3aee4eeefee4">

### 
Slight spacing improvements and removed paddings. I could make it so that only the table will scroll and not the whole modal/popup content but will need to make a change in the base package I think.

<img width="1207" alt="Screenshot 2024-04-01 at 23 11 18" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/d65600a9-be9b-437d-adf5-fe6f26df5bc7">

If there is anything you don't like let me know and I'll take a look.

parallels999 commented on src/Resources/laravel-debugbar.css:
```diff
@@ -352,14 +375,25 @@ ul.phpdebugbar-widgets-timeline table.phpdebugbar-widgets-params {
+}
+
 div.phpdebugbar-widgets-messages div.phpdebugbar-widgets-toolbar {
-    width: calc(100% - 20px);
```
Normally these `-20 px` were added when there is a lot of data the scrollbar appears, then a presentation problem occurred in the toolbar, maybe is a regression

parallels999 replied:
Perhaps many of these changes should be applied first to [maximebf/php-debugbar](https://github.com/maximebf/php-debugbar)

nckrtl replied:
Do you know perhaps of a specific tab that caused scrolling issues? And we're talking about an unwanted horizontal scrollbar right? I did get a horizontal scrollbar with the change you marked but I applied more changes to get rid of it.

I can certainly extract changes to the base package. I was just so far in with the changes that I thought it might make more sense to first gather feedback and then extract (if people would agree with it).

parallels999 commented:
>Updated spacing and line height of the lines in the error for bit better readability

![image](https://github.com/barryvdh/laravel-debugbar/assets/109294935/986e67bc-0e20-4c8f-afe5-32cfb4728a7c)
You break it

parallels999 commented:
>The percentages bars were being displayed with table borders but I thought it would make sense to diplay them in light/dark background as the lines in Requests tabs are displayed as well. Making it a bit more consistent.

In this way, the final results can no longer be differentiated much from the timeline waterfall.
They can be confused as concurrent
![image](https://github.com/barryvdh/laravel-debugbar/assets/109294935/d790370d-f249-4c90-b548-e1ef6d4d53df)

parallels999 commented:
With specific changes in different PRs it would be easier to review

nckrtl commented:
> > Updated spacing and line height of the lines in the error for bit better readability
> 
> ![image](https://private-user-images.githubusercontent.com/109294935/318587812-986e67bc-0e20-4c8f-afe5-32cfb4728a7c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwNDIzMjYsIm5iZiI6MTcxMjA0MjAyNiwicGF0aCI6Ii8xMDkyOTQ5MzUvMzE4NTg3ODEyLTk4NmU2N2JjLTBlMjAtNGM4Zi1hZmU1LTMyY2ZiNDcyOGE3Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMlQwNzEzNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYmQ5ZDg1N2NlYzhhN2I1NDk5ODVkMWYwYzIwZDEwMDc4ZmNhZGJhM2M5Y2I1MWJiOTI0YzJiNzE2NTU4OWRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.xT544YP6LAarv8M3yZIEkwMTbgjU9kKIhWeF8MS8XjA) You break it

Fixed
<img width="698" alt="Screenshot 2024-04-02 at 09 00 07" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/80629bab-2a9a-45cc-b91a-5c00ba160b2d">

### 
### 

> > The percentages bars were being displayed with table borders but I thought it would make sense to diplay them in light/dark background as the lines in Requests tabs are displayed as well. Making it a bit more consistent.
> 
> In this way, the final results can no longer be differentiated much from the timeline waterfall. They can be confused as concurrent(yes could be concurrent, try views timeline) ![image](https://private-user-images.githubusercontent.com/109294935/318588309-d790370d-f249-4c90-b548-e1ef6d4d53df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwNDIzMjYsIm5iZiI6MTcxMjA0MjAyNiwicGF0aCI6Ii8xMDkyOTQ5MzUvMzE4NTg4MzA5LWQ3OTAzNzBkLWYyNDktNGM5MC1iNTQ4LWUxZWY2ZDRkNTNkZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMlQwNzEzNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMjAzYjNlZTVlODMzNGM0MTlmNjBmNzJkYWRlOGI0ZDQxNmQ4NWY5ZmQxMGU4OWQwYjY1NjlkZTZiNWNmODI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.0KCP7FyhJXffmGKdjQHcuoFKd_ruMb_AxPiTfgawXoI)

Brought back table borders:
<img width="1260" alt="Screenshot 2024-04-02 at 09 24 28" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/fa358183-fa9d-4179-97ff-b106818f6f05">

### 
### 

> With specific changes in different PRs it would be easier to review

I could separate the changes in different PR's. Will need to extract some changes to the base package anyways as you mentioned that'd make more sense. But will do that once it's clear that these changes are considered an improvement. Waiting for more feedback for now.

barryvdh commented:
I really like the query change, so that it's not the entire background :)

There seems to be some weird spacing if an item has multiple lines. Also visible on your screenshot, in the request_headers row.


> On the requests tab the background of the key items was not extending to the bottom of the of value items. This is now fixed and look a bit cleaner I think.
> 
> <img alt="Screenshot 2024-04-01 at 23 10 47" width="1488" src="https://private-user-images.githubusercontent.com/18613261/318571919-4c7dfb98-9a62-4da1-affd-7f63aa5dd206.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwNDkwODYsIm5iZiI6MTcxMjA0ODc4NiwicGF0aCI6Ii8xODYxMzI2MS8zMTg1NzE5MTktNGM3ZGZiOTgtOWE2Mi00ZGExLWFmZmQtN2Y2M2FhNWRkMjA2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAyVDA5MDYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2NzU0ODBkODY0YjYwZjM1M2ViNDc5NGY5NTc5YmY0NzIwZThhYTRlZjcyNDU0ZGE4NDBkNmYyZjQ4YTM4ZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yAxW74irLFnh3j4NF3YOD-KuLyXjHEueQ1aMduCpfGQ">

Same with multiple middleware:
<img width="735" alt="image" src="https://github.com/barryvdh/laravel-debugbar/assets/973269/5dce3aa4-ecc0-4d30-a6cf-5630cabca54b">


I see you changed the icons, but the close and browse icons still feel 'fuzzy' vs the other icons
<img width="209" alt="image" src="https://github.com/barryvdh/laravel-debugbar/assets/973269/79c8f586-6288-42c2-a146-239eedefa498">

Not sure if that was intended to change or not.

nckrtl commented:
Thanks for the feedback and giving it a spin. Regarding the spacing you mean these spacings/indent right?
<img width="1488" alt="318571919-4c7dfb98-9a62-4da1-affd-7f63aa5dd206-2" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/9ade5f02-cb75-4974-a993-b68074888376">
Seems it also has weird indents/spacing in the current version but my changes seem to have intensified the spacing. Will take a look at it. The middleware example is clear, will try and reproduce this.

I didn't change the icons actually. I adjusted the positioning but as the change happened on the same line as the base64 encoded value the whole line is seen by git to be updated I think. The fuzziness is being caused by the icons being base64 enconded PNG's, not SVGs. I can replace them for svg's which should solve the fuzziness.

barryvdh commented:
Yeah I mean those indents.

I'll look at the icons myself then, maybe I can just fix it upstream.

Also seeing a lot of '!important` overrides, maybe it would be easier to change that in https://github.com/maximebf/php-debugbar/blob/master/src/DebugBar/Resources/debugbar.css ?

nckrtl commented:
Yeah will give it a try moving some of the changes to that package. Will take a look at it later tonight. To be continued :)

barryvdh commented:
Thanks!
For icons, I created https://github.com/maximebf/php-debugbar/pull/644 which replaces the svgs with font-awesome icons, so it's more inline with the rest (and easier to change color)

nckrtl commented:
> Thanks for the feedback and giving it a spin. Regarding the spacing you mean these spacings/indent right? <img alt="318571919-4c7dfb98-9a62-4da1-affd-7f63aa5dd206-2" width="1488" src="https://private-user-images.githubusercontent.com/18613261/318725587-9ade5f02-cb75-4974-a993-b68074888376.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxMzEwOTksIm5iZiI6MTcxMjEzMDc5OSwicGF0aCI6Ii8xODYxMzI2MS8zMTg3MjU1ODctOWFkZTVmMDItY2I3NS00OTc0LWE5OTMtYjY4MDc0ODg4Mzc2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDA3NTMxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlNjFlODUyYTIwNzMyNjA3ZTY2N2ViOGRhOTk2NmVmZTVlNDZjZGViYjk5NTEwOGUzMjg0MWIxYmFlNzQ0NDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gPA9kGzclhJDSgbJs-6qaUZBD-x42_vN1HVYGvwXZgs"> Seems it also has weird indents/spacing in the current version but my changes seem to have intensified the spacing. Will take a look at it. The middleware example is clear, will try and reproduce this.
> 
> I didn't change the icons actually. I adjusted the positioning but as the change happened on the same line as the base64 encoded value the whole line is seen by git to be updated I think. The fuzziness is being caused by the icons being base64 enconded PNG's, not SVGs. I can replace them for svg's which should solve the fuzziness.

Fixed, only applies to Laravel debugbar. It was caused by a width/margin/padding combination of value and key element.

<img width="1045" alt="Screenshot 2024-04-03 at 09 50 40" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/dd54012c-dfb2-4656-9c4b-1fbc580e5bc3">

barryvdh commented:
Thanks! Looks better.  Should I merge #1603 first or is that conflicting?

I also see that the active/highlight color is too short, but I think that was already the case 🤔 
![image](https://github.com/barryvdh/laravel-debugbar/assets/973269/aaffc2c9-5c3f-4d9b-927f-6478725f388e)

nckrtl commented:
If you merge https://github.com/barryvdh/laravel-debugbar/pull/1603 il take a look at the active/highlight height in this PR

barryvdh commented:
Okay merged it :)

nckrtl commented:
My latest commit is a bit messy but did the following things:
- Replaced the Laravel icon for an SVG version, just to be more consistent with the other icon formats
- Fixed the active tab height. Also no more gap between bottom header bar and other header items/tabs:
<img width="839" alt="Screenshot 2024-04-03 at 11 51 18" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/1edb4660-536a-437e-9c57-31fddf7c4fe5">
- Also applied some more alignment of the logo to better align with tab content
- Text is also visually more centered in the tabs in combination with the pills/counters

barryvdh commented:
Hmm it slight different with me. This is with dev-master of php-debugbar right?
![image](https://github.com/barryvdh/laravel-debugbar/assets/973269/a06b80bd-6e76-4c2b-8633-9952bffbba71)

On the right side it's correct
![image](https://github.com/barryvdh/laravel-debugbar/assets/973269/8b33d96d-73fe-44f0-b734-9a79f8758584)

Only the dropdown is a bit crushed

nckrtl commented:
@barryvdh Yeah, looking good for me in chrome and safari. What browser are you using? Do you still have the issue? I pushed some updates to better align the dropdown in different browsers

barryvdh commented:
The dropdown is because of Filament, but I can fix that by forcing the line-height to 1em on the select.

The tabs are still a bit weird, but I can add height:100% to them and it's fixed.

Rest looks good, is this ready to go?

nckrtl commented:
Alright. Yeah about the dropdown, I had something similar with a specific project that had styling for select elements. The base package could benefit from having dedicated select styling, so other projects won't affect it. Will look into it.

I think its ready to go yeah :)

barryvdh commented:
Thanks, fixed the height in https://github.com/maximebf/php-debugbar/commit/424be4f885f278edddf4ffa2e3dd41dc04279096
Not sure why the 16px was needed, because it conflicted for me and doesn't seem to be required.

barryvdh commented:
Looks good. It's tagged now on both versions just let's see what pops up :)

nckrtl commented:
> Thanks, fixed the height in [maximebf/php-debugbar@424be4f](https://github.com/maximebf/php-debugbar/commit/424be4f885f278edddf4ffa2e3dd41dc04279096) Not sure why the 16px was needed, because it conflicted for me and doesn't seem to be required.

It was added to remove a 1px gap between some of the tab items and the bottom border of the header but your change also seems to do the trick, looking good at my end :)

If anything arises feel free to tag me, happy to take a look.

erikn69 commented on src/Resources/laravel-debugbar.css:
```diff
@@ -579,6 +597,10 @@ ul.phpdebugbar-widgets-list li.phpdebugbar-widgets-list-item .phpdebugbar-widget
 }
 
+ul.phpdebugbar-widgets-list li.phpdebugbar-widgets-list-item {
+    line-height: 20px;
```
After this change seems like messages list grows too much on gate tab, and the icons are misaligned
![image](https://github.com/barryvdh/laravel-debugbar/assets/4933954/9311b75a-7498-4dd4-a680-24bbd739073b)
Same on messages tab
![image](https://github.com/barryvdh/laravel-debugbar/assets/4933954/566af5e7-b827-49a0-aadc-eee76220d0c1)

Why don't just normal line height??

erikn69 replied:
Try #1604

erikn69 commented on src/Resources/laravel-debugbar.css:
```diff
@@ -345,14 +356,26 @@ ul.phpdebugbar-widgets-timeline table.phpdebugbar-widgets-params {
+}
+
 div.phpdebugbar-widgets-messages div.phpdebugbar-widgets-toolbar {
-    width: calc(100% - 20px);
```
This is a regression, 
the `-20px` was there to prevent the toolbar from obstructing the vertical scrollbar, on all the tabs extending messages

![image](https://github.com/barryvdh/laravel-debugbar/assets/4933954/44f93b5b-8cb9-425e-ba81-e59ab81ff637)

![image](https://github.com/barryvdh/laravel-debugbar/assets/4933954/07e06b1f-22db-4d37-9e25-5b5cddfd9147)

nckrtl replied:
The same goes for the close button then? I expect you have this issue as well with the base package? http://phpdebugbar.com. On Mac the scrollbars hide or are half translucent when scrolling so I didn't encountered this. Is this IE or something?

nckrtl commented on src/Resources/laravel-debugbar.css:
```diff
@@ -307,13 +314,17 @@ dl.phpdebugbar-widgets-kvlist dd.phpdebugbar-widgets-value.phpdebugbar-widgets-p
 
 dl.phpdebugbar-widgets-kvlist dt {
-    width: 25%;
+    width: calc(25%-10px);
```
I thought it helped to better align these items:
> Fixed, only applies to Laravel debugbar. It was caused by a width/margin/padding combination of value and key element.
> 
> <img alt="Screenshot 2024-04-03 at 09 50 40" width="1045" src="https://private-user-images.githubusercontent.com/18613261/319092581-dd54012c-dfb2-4656-9c4b-1fbc580e5bc3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNjA0MjMsIm5iZiI6MTcxMjE2MDEyMywicGF0aCI6Ii8xODYxMzI2MS8zMTkwOTI1ODEtZGQ1NDAxMmMtZGZiMi00NjU2LTljNGItMWZiYzU4MGU1YmMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE2MDIwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2NDEyOThkNmJhODkyMWUwM2U0YjgzYTIxNWJhMzc2MzNlMmFhNTNmOTlhZTcxZTM1NmJjNWY0YmY1NzcyYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IkrKqueDM-Enr3u9gztk4XeVOYszuqEPMdxRqw4cRjQ">

But you're right that's not doing anything indeed, will put it on the list to undo.

erikn69 replied:
It seems that this should not be removed

https://github.com/barryvdh/laravel-debugbar/pull/1605#discussion_r1551759026

barryvdh commented:
So I've merged both your PR's. Anything missing?

nckrtl commented:
Not from my end, all good. Are the issue fixed for you @parallels999 and @erikn69 ?

nckrtl commented:
Hmm can't seem to reproduce this, what looks like for me on the latest version and on master:

<img width="431" alt="Screenshot 2024-04-03 at 19 45 35" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/1cab2e1c-ca65-46f5-a0e2-331e6753ec4e">
<img width="342" alt="Screenshot 2024-04-03 at 19 45 29" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/8e706c2b-dd88-4256-b8f7-5ed37d30076f">

On page load I do see "View in Telescope" in blue for a split second and then it goes to the correct color.

nckrtl commented:
Do you have it in other apps as well. All my local apps seem to render the texts correctly

erikn69 commented:
> I do see that the color is off, not sure why that is:

On some places, me too, there is a new text shadow or is it my imagination?

erikn69 commented on src/Resources/laravel-debugbar-dark-mode.css:
```diff
@@ -31,8 +31,11 @@ div.phpdebugbar code.phpdebugbar-widgets-sql span.hljs-keyword,
 div.phpdebugbar-openhandler .phpdebugbar-openhandler-header,
 div.phpdebugbar-openhandler .phpdebugbar-openhandler-header a {
     color: var(--color-gray-200);
+    text-shadow: 1px 1px #000;
```
Maybe here??

nckrtl replied:
Yeah I added that to make the text in the timeline waterfall better readable on blue background:
<img width="699" alt="Screenshot 2024-04-03 at 20 22 30" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/77639078-bf56-4808-bc1c-a5b1b0ae6994">

With the text shadow:
<img width="861" alt="Screenshot 2024-04-03 at 20 28 27" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/662f5df4-c40b-460d-9c8c-42f1569402d9">

But quite some elements are affected by that change, perhaps too many:
<img width="861" alt="Screenshot 2024-04-03 at 20 28 27" src="https://github.com/barryvdh/laravel-debugbar/assets/18613261/d1aaddd9-3a62-4e74-a5e3-01f1f8cb4de3">

erikn69 replied:
>I added that to make the text in the timeline waterfall better readable

Since there are many classes involved(_also main `div.phpdebugbar`_), I saw it everywhere.

barryvdh commented:
Ah the color was because of quirks mode; https://stackoverflow.com/questions/24258380/font-color-in-a-table-doesnt-inherit-from-parent-div
https://github.com/maximebf/php-debugbar/pull/646

barryvdh commented:
Hmm no, is that because of the color inherit?

erikn69 commented:
>is that because of the color inherit?

No, it seems that it was already there, it happens in dark theme

erikn69 commented:
>is that because of the color inherit?

No, it seems that it was already there, it happens in dark theme

![image](https://github.com/barryvdh/laravel-debugbar/assets/4933954/d26910ae-b766-4992-93f4-71eaf00ff29a)

If you add `color: white;` there
![image](https://github.com/barryvdh/laravel-debugbar/assets/4933954/1d276767-20d5-4b8c-8639-1ae8388e34e8)

parallels999 commented:
Still not 100% aligned
![image](https://github.com/barryvdh/laravel-debugbar/assets/109294935/0902dbc7-db02-4d1a-9731-7a145eb46f50)
